accepted/2025/prune-package-reference-rollout.md

```diff
@@ -0,0 +1,70 @@
+- We will continuously monitor telemetry and feedback around the feature through all the previews.
+
+### Testing & Validation
+
+The feature has been development since November and the implementation has been extensively tested through automation, but we also ensured that our own repos can dogfood without any issues.
+
+We validated the .NET SDK build that it *does not affect the builds negatively of the following repos*:
+
+- NuGet/NuGet.Client
+- dotnet/sdk
+- dotnet/roslyn - <https://github.com/dotnet/roslyn/pull/76898>
+- dotnet/runtime - <https://github.com/dotnet/runtime/pull/111767>
+- dotnet/aspnetcore - <https://github.com/dotnet/aspnetcore/pull/60025>
+
+### Observable build changes
+
+- Customers who expect the output of their project to produce identical results will notice smaller deps files. This is intentional.
+- Customers who observe the output of component governance / NuGet Audit will notice these components are no longer reported. This is intentional.
+- Customers who use packages lock files (1% usage) with locked mode may run into restore failures due to restore bringing in fewer dependencies. This is intentional.
+- Customers with builds that have customization may run into build failures if they are:
+  - Manually referencing an assembly or props/targets from a package from the global packages folder. Note that pruning is not allowed for direct dependencies, so this would need to be a transitive package.
+    - Manual assembly coping - at runtime, the platform version will be preferred anyways, but it could lead to the build failing if the package cannot be found on disk anymore.
+  - We don't expect this to be a common scenario given that the packages being pruned are within the framework and customers do not need do to anything custom to make things work.
+
+## Rationale and alternatives
+
+- Roll-out the feature based on the target framework.
+  - Enable pruning for .NET 10 frameworks only.
+  - Enable pruning for all frameworks of projects that target .NET 10 (ie. multi-targetting .NET 10 and NET 8.0 in a project enables pruning for both).
+- Enable pruning when NuGetAuditMode is set to `all` - reducing transitive false positives for audit.
+
+## Prior Art
+
+- Pruning is the restore equivalent of [the build time conflict resolution](<https://github.com/dotnet/sdk/blob/262b9c3d6cf67287f649e38d83e6c5d9d08feb8a/src/Tasks/Common/ConflictResolution/ResolvePackageFileConflicts.cs#L178-L182>), that is current enable for every framework.
+
+## Unresolved Questions
+
+<!-- What parts of the proposal do you expect to resolve before this gets accepted? -->
+<!-- What parts of the proposal need to be resolved before the proposal is stabilized? -->
+<!-- What related issues would you consider out of scope for this proposal but can be addressed in the future? -->
+
+## Future Possibilities
+
+- Enabling pruning for .NET Framework.
```

Comment:
Just curious, we're making this call based on telemetry.  Is it worth calling out how many restores will be left out if we don't include .NET Framework?

Reply:
We don't know that unfortunately.